### PR TITLE
Feature: Inline guest

### PR DESCRIPTION
**Pull request by khanghoang**
- Add animations for inline guest.
- Add localizations.

### CODE REVIEW

**Comment by bluepnume** [comment]
Can you remove the /dist files from this PR?

File: src/billing/index.js [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+    name: 'billing-page',
+
+    buildUrl(props) : string {
+        let env = props.env || config.env;
+        let { lang, country } = config.locale || getBrowserLocale();
+        const locale = `${ lang }_${ country }`;
+        let isCommit = props.commit ? 1 : 0;
+        const { token } = props;
+
+        return `${ config.inlinedCardFieldUrls[env] }/billing?locale.x=${ locale }&commit=${ isCommit }&token=${ token }`;
```
Comment (bluepnume) [inline src/billing/index.js]: We shouldn't put anything in the url here. This should all be handled using props.

File: src/billing/index.js [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+    },
+
+    get domain() : Object {
+        return {
+            ...config.paypalDomains,
+            [ ENV.LOCAL ]: /^http:\/\/localhost.paypal.com:\d+$/
+        };
+    },
+
+    get bridgeUrl() : Object {
```
Comment (bluepnume) [inline src/billing/index.js]: We can remove this. No popup window so no bridge needed.

File: src/billing/index.js [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+        return config.metaFrameUrls;
+    },
+
+    get bridgeDomain() : Object {
```
Comment (bluepnume) [inline src/billing/index.js]: Can remove this.

File: src/billing/template/containerTemplate.jsx [hunk 4]
```diff
@@ -0,0 +1,131 @@
+/* @flow */
```
Comment (bluepnume) [inline src/billing/template/containerTemplate.jsx]: Is there anything different here from the checkout overlay? If so can we have that as a common function between the two components?

File: src/button/config.js [hunk 5]
```diff
@@ -516,6 +516,8 @@ export const BUTTON_RELATIVE_STYLE = {
     VERTICAL_MARGIN: 30
 };
 
+export const POWERED_BY_PAYPAL_HEIGHT = 30;
```
Comment (bluepnume) [inline src/button/config.js]: Can we put this in the `BUTTON_STYLE` config?

File: src/card/index.js [hunk 6]
```diff
@@ -19,53 +20,77 @@ type CardOptions = {
...
         let env = props.env || config.env;
+        let { lang, country } = config.locale || getBrowserLocale();
+        const locale = `${ lang }_${ country }`;
+        const { token } = props;
+        let isCommit = props.commit ? 1 : 0;
 
-        return window.xprops.payment().then(paymentToken => {
-            return `${ config.inlinedCardFieldUrls[env] }?token=${ paymentToken }`;
-        });
+        return `${ config.inlinedCardFieldUrls[env] }?token=${ token }&locale.x=${ locale }&commit=${ isCommit }`;
```
Comment (bluepnume) [inline src/card/index.js]: Same here -- let's use props rather than building the url props manually.

File: test/screenshot/screenshot.test.js [hunk 7]
```diff
@@ -47,12 +47,17 @@ afterAll(async () => {
...
+let index = 1;
+
 for (let config of buttonConfigs) {
     let filename = config.filename || dotifyToString(config) || 'base';
 
     test(`Render button with ${ filename }`, async () => {
         let { page } = await setupBrowserPage;
 
+        // fasten up the animation
+        await page._client.send('Animation.setPlaybackRate', { playbackRate: 12.0 });
```
Comment (bluepnume) [inline test/screenshot/screenshot.test.js]: Awesome

File: test/screenshot/screenshot.test.js [hunk 8]
```diff
@@ -117,5 +122,7 @@ for (let config of buttonConfigs) {
         } else {
             await screenshot.write(filepath);
         }
+        index += 1;
+        console.log(`${ index } / ${ total }`);
```
Comment (bluepnume) [inline test/screenshot/screenshot.test.js]: `console.log(`Generating button screenshot: ${ index } / ${ total }`);`

**Comment by bluepnume** [comment]
Still need to remove /dist changes please.

File: src/billing/index.js (hunk 1 above)
Comment (bluepnume) [reply src/billing/index.js]: Still need to remove query params here.

File: src/billing/index.js [hunk 9]
```diff
@@ -0,0 +1,146 @@
...
+            queryParam: true
+        },
+
+        commit: {
+            type:       'boolean',
+            required:   false,
+            queryParam: true
+        },
+
+        fundingSource: {
```
Comment (bluepnume) [inline src/billing/index.js]: Is this used? Wouldn't this always be `card`?

File: src/billing/template/containerTemplate.jsx (hunk 4 above)
Comment (bluepnume) [reply src/billing/template/containerTemplate.jsx]: Looks like this is still duplicated?

File: src/hacks.js [hunk 10]
```diff
@@ -225,3 +226,7 @@ if (Button.isChild()) {
         window.Promise = ZalgoPromise;
     }
 }
+
+patchMethod(BillingPage, 'renderTo', ({ args: [ win, props, el ], original, context }) => {
```
Comment (bluepnume) [inline src/hacks.js]: I think we can remove this. Just call `renderTo(window.top, ...)` when you render.

**Comment by khanghoang** [comment]
@bluepnume I've updated PR, please review it again.

File: src/billing/template/containerTemplate.jsx (hunk 4 above)
Comment (khanghoang) [reply src/billing/template/containerTemplate.jsx]: oh, the UI of billing address popup is slightly different with the checkout so I reuse whatever I can.

File: src/button/template/containerTemplate.jsx [hunk 11]
```diff
@@ -64,6 +64,10 @@ function getDimensions({ label, size, tagline, fundingicons, layout, number, vie
         height = (buttonHeight * number) + (perc(buttonHeight, BUTTON_RELATIVE_STYLE.VERTICAL_MARGIN) * (number - 1));
     }
 
+    if (layout === BUTTON_LAYOUT.VERTICAL && (size === BUTTON_SIZE.LARGE || size === BUTTON_SIZE.MEDIUM)) {
```
Comment (bluepnume) [inline src/button/template/containerTemplate.jsx]: Shouldn't we check `cards.length` here so the check is the same as in the template?

Comment (khanghoang) [reply src/button/template/containerTemplate.jsx]: what's a template you're talking about? the `xo-button` one?

Comment (bluepnume) [reply src/button/template/containerTemplate.jsx]: I mean here: https://github.com/paypal/paypal-checkout/pull/758/files#diff-4957c6fdc3594ef3bb401ab5307f46d2R349

File: src/button/template/componentTemplate.jsx [hunk 12]
```diff
@@ -257,6 +262,40 @@ function renderStyle({ height, cardNumber } : { height? : ?number, cardNumber? :
...
 
+function renderPowerByPaypalLogo(props) : ChildType {
+
+    if (!props) {
+        return null;
+    }
+
+    const { layout, size } = props;
+
+    if (!(layout === BUTTON_LAYOUT.VERTICAL && (size === BUTTON_SIZE.MEDIUM || size === BUTTON_SIZE.LARGE || size === BUTTON_SIZE.HUGE))) {
```
Comment (bluepnume) [inline src/button/template/componentTemplate.jsx]: Just check vertical, remove size checks

**Pull request merged**